modules/masterbar/masterbar.php

```diff
@@ -0,0 +1,821 @@
+
+		return str_replace( '/', '::', $url );
+	}
+
+	function wpcom_static_url( $file ) {
+		$i   = hexdec( substr( md5( $file ), - 1 ) ) % 2;
+		$url = 'http://s' . $i . '.wp.com' . $file;
+
+		return set_url_scheme( $url );
+	}
+
+	public function replace_core_masterbar() {
+		global $wp_admin_bar;
+
+		if ( ! is_object( $wp_admin_bar ) ) {
+			return false;
+		}
+
+		$this->clear_core_masterbar( $wp_admin_bar );
+		$this->build_wpcom_masterbar( $wp_admin_bar );
+	}
+
+	// Remove all existing toolbar entries from core Masterbar
+	public function clear_core_masterbar( $wp_admin_bar ) {
+		foreach ( $wp_admin_bar->get_nodes() as $node ) {
+			$wp_admin_bar->remove_node( $node->id );
+		}
+	}
+
+	// Add entries corresponding to WordPress.com Masterbar
+	public function build_wpcom_masterbar( $wp_admin_bar ) {
+		// Menu groups
+		$this->wpcom_adminbar_add_secondary_groups( $wp_admin_bar );
+
+		// Left part
+		$this->add_my_sites_submenu( $wp_admin_bar );
+		$this->add_reader_submenu( $wp_admin_bar );
+
+		// Right part
+		if ( Jetpack::is_module_active( 'notes' ) ) {
+			$this->add_notifications( $wp_admin_bar );
+		}
+
+		$this->add_me_submenu( $wp_admin_bar );
+		$this->add_write_button( $wp_admin_bar );
+	}
+
+	public function get_locale() {
+		$wpcom_locale = get_locale();
+
+		if ( ! class_exists( 'GP_Locales' ) ) {
+			if ( defined( 'JETPACK__GLOTPRESS_LOCALES_PATH' ) && file_exists( JETPACK__GLOTPRESS_LOCALES_PATH ) ) {
+				require JETPACK__GLOTPRESS_LOCALES_PATH;
+			}
+		}
+
+		if ( class_exists( 'GP_Locales' ) ) {
+			$wpcom_locale_object = GP_Locales::by_field( 'wp_locale', get_locale() );
+			if ( $wpcom_locale_object instanceof GP_Locale ) {
+				$wpcom_locale = $wpcom_locale_object->slug;
+			}
+		}
+
+		return $wpcom_locale;
+	}
+
+	public function add_notifications( $wp_admin_bar ) {
+		$wp_admin_bar->add_node( array(
+			'id'     => 'notes',
+			'title'  => '<span id="wpnt-notes-unread-count" class="wpnt-loading wpn-read"></span>
+						 <span class="noticon noticon-bell"></span>',
+			'meta'   => array(
+				'html'  => '<div id="wpnt-notes-panel2" style="display:none" lang="'. esc_attr( $this->locale ) . '" dir="' . ( $this->is_rtl() ? 'rtl' : 'ltr' ) . '">' .
+				           '<div class="wpnt-notes-panel-header">' .
+				           '<span class="wpnt-notes-header">' .
+				           __( 'Notifications', 'jetpack' ) .
+				           '</span>' .
+				           '<span class="wpnt-notes-panel-link">' .
+				           '</span>' .
+				           '</div>' .
+				           '</div>',
+				'class' => 'menupop',
+			),
+			'parent' => 'top-secondary',
+		) );
+	}
+
+	public function add_reader_submenu( $wp_admin_bar ) {
+		$wp_admin_bar->add_menu( array(
+			'id'    => 'newdash',
+			'title' => __( 'Reader', 'jetpack' ),
+			'href'  => 'https://wordpress.com/',
+		) );
+
+		$wp_admin_bar->add_menu( array(
+			'parent' => 'newdash',
+			'id'     => 'streams-header',
+			'title'  => __( 'Streams', 'jetpack' ),
```

Comment:
This might benefit from a translator comment, because "Streams" doesn't mean much and is hard to translate without context.